pkg/chunk/aws_storage_client.go

```diff
@@ -298,6 +306,7 @@ func (a awsStorageClient) QueryPages(ctx context.Context, query IndexQuery, call
 
 		backoff = minBackoff
 	}
+	dynamoQueryPagesCount.Observe(float64(pageCount))
```

Comment:
There are a couple of early returns in the loop, so this `Observe` call won't always happen.